### ISSUE TITLE
Reduction Op Perf Model

### ISSUE THREAD

**Issue by ipotkonjak-tt** [tenstorrent/tt-mlir#1329]
### Goal
We aim to provide a model of the reduction op performance based on the possible combinations of input parameters. Because of the complexity of the op in question we will focus on modelling a simple control flow path (that excludes other op calls within reduction). First goal is to provide modelling PoC.

We will attempt to model the perf with simpler functions, like Linear Regression, and move on to more complex ones if there is a need. Training data will be split into logical segments and for each of them a different model will be created. For each of the models accuracy error will be provided, to give Optimizer information about the quality of the estimate.

Additional information and results will be provided in the comment section and sub-issues.
[Op Perf Model - Parent Issue](https://github.com/tenstorrent/tt-mlir/issues/305)

**Comment by ipotkonjak-tt** [comment]
Modelling showed height reduction perf variations, more details can be found in the [related issue](https://github.com/tenstorrent/tt-metal/issues/15498).

**Comment by skhatriTT** [comment]
Hey @ipotkonjak-tt what is our status for this issue?

**Comment by ipotkonjak-tt** [comment]
Hey @skhatriTT, PoC model was made for the non composite reduction op. You can find that analysis in the following [doc](https://docs.google.com/document/d/1lh9fQMy-7Ac3tASLE1E20u-MEY8aHldmnVsAXnRPXxg/edit?tab=t.0). 

After that I've been moved to another project so don't know the current status.
